backend/service/planInvite.ts

```diff
@@ -0,0 +1,70 @@
+import { type User, InviteStatus, Role } from '@prisma/client';
+import PlanMembershipModel from '../models/planMembership';
+import PlanInviteModel from '../models/planInvite';
+
+export default class PlanInviteService {
+  static async setUpInviteePlanMembership(planInviteID: string, user: User): Promise<string> {
+    const planInviteData = await PlanInviteModel.readOnePlanInvite({ id: planInviteID })
+
+    if (planInviteData === null) throw new Error('fail to fetch planID from planInvite model');
+
+    // create plan membership
+    await PlanMembershipModel.dbCreateOnePlanMembership({
+      planID: planInviteData.planID,
+      userID: user.id,
+      inviteStatus: InviteStatus.ACCEPTED,
+      role: Role.FRIEND
+    })
+
+    // return plan id
+    return planInviteData.planID
+  }
+
+  static async isUserAlreadyPlanMember(planInviteID: string, userID: string) {
+    const planInviteData = await PlanInviteModel.readOnePlanInvite({
+      id: planInviteID
+    })
+
+    if (planInviteData === null) throw new Error('unable to find plan')
+
+    const isUserInPlan = await PlanMembershipModel.dbReadOnePlanMembership({ planID: planInviteData.planID, userID })
+
+    // if no record is found, then user is not in plan
+    if (isUserInPlan === null) return false
+
+    return true
```

Comment:
maybe less readable, so no change needed, but i'd say we could `return !!isUserInPlan`

Reply:
This is the first time i've seen this syntax